pipeline/mutate/mutator_id_token.go

```diff
@@ -108,16 +104,21 @@ func (a *MutatorIDToken) tokenFromCache(config *CredentialsIDTokenConfig, sessio
 }
 
 func (a *MutatorIDToken) tokenToCache(config *CredentialsIDTokenConfig, session *authn.AuthenticationSession, claims []byte, ttl time.Duration, expiresAt time.Time, token string) {
-	if !a.tokenCacheEnabled {
+	if !config.Cache.Enabled {
 		return
 	}
 
 	key := a.cacheKey(config, ttl, claims, session)
-	a.tokenCache.Set(key, &idTokenCacheContainer{
-		TTL:       ttl,
-		ExpiresAt: expiresAt,
-		Token:     token,
-	}, 0)
+	a.tokenCache.SetWithTTL(
+		key,
+		&idTokenCacheContainer{
+			TTL:       ttl,
+			ExpiresAt: expiresAt,
+			Token:     token,
+		},
+		0,
+		ttl,
```

Comment:
In other caches, I believe we set the TTL to `min(TTL, time.Until(expiresAt)`. That would make sense here too IMO.

Reply:
They should be same normally. Since calling this method does:
```
	now := time.Now().UTC()
	exp := now.Add(ttl)
	[...]
	a.tokenToCache(c, session, templateClaims, ttl, exp, signed)
```

Or what am I missing? :)

Reply:
LGTM